### PR TITLE
chore: auto update dev dependencies

### PR DESCRIPTION
**Pull request by DiegoRBaquero**
Also, don't auto update major types (to match installed package version)

### CODE REVIEW

File: renovate.json [hunk 1]
```diff
@@ -39,5 +39,16 @@
...
   },
   "packageRules": [
+    {
+      "description": "Auto update dev dependencies",
+      "depTypeList": ["devDependencies"],
+      "automerge": true
+    },
+    {
+      "description": "Auto update non-major types dev dependencies",
+      "packagePatterns": ["^@types/"],
```
[the remark this answers is not in the document]
Comment (DiegoRBaquero) [reply renovate.json:49]: PR has matching minor and patch, that's why major is excluded, but you only want patch?

[the remark this answers is not in the document]
Comment (aorinevo) [reply renovate.json:49]: The issue is in maintaining parity between a dependency and its types.  For example, in dependencies we have `js-yaml` and `@types/js-yaml`.  We want to keep these two in-sync on minor.  That is, if `@types/js-yaml` is on v3.14.x if-and-only-if `js-yaml` is on v3.14.y for some non-negative integers x and y.

With this PR, `@types` can fall out of sync with the corresponding dependency.

[the remark this answers is not in the document]
Comment (aorinevo) [reply renovate.json:49]: That said, we have a bunch of dependencies out of parity on master.  So, if anything, this change will likely help bring some of these back to parity up to minor.

[the remark this answers is not in the document]
Comment (aorinevo) [reply renovate.json:49]: I believe there is more benefit to landing this as-is to help close the existing parity issues.  We can leverage manual approvals via renovatebot to keep dependencies and @types in sync.

**Pull request merged**